common/data.tf

```diff
@@ -183,9 +201,14 @@ resource "null_resource" "deploy_hieradata" {
   provisioner "remote-exec" {
     inline = [
       "sudo mkdir -p /etc/puppetlabs/data",
+      "sudo mkdir -p /etc/puppetlabs/facts",
       "sudo install -m 650 terraform_data.yaml user_data.yaml /etc/puppetlabs/data/",
+      "sudo install -m 650 terraform_facts.yaml /etc/puppetlabs/facts/",
+      # These chgrp commands do nothing if the puppet group does not yet exist
+      # so these are also handled by puppetmaster.yaml
```

Comment:
For clarification, the reason why chgrp is here too is because the provisioner can be applied more than once and at any point of the cluster lifetime.

Reply:
Should I remove the comments? I added them because they led to some confusion for me

Reply:
I think you were right to add the comment, no need to remove it. I just wanted to explain why it is done twice.